replica/table.cc

```diff
@@ -1962,6 +1962,8 @@ void table::handle_tablet_split_completion(size_t old_tablet_count, const locato
                                               "therefore groups cannot be remapped with the new tablet count.",
                                               id, table_id));
         }
+        // Remove old main groups, they're unused, but they need to be deregistered properly
+        co_await sg->main_compaction_group()->stop();
         unsigned first_new_id = id << growth_factor;
```

Comment:
we cannot yield here, the remapping must be atomic. ideally, we don't futurize handle_tablet_split_completion, so we make it explicit.

Reply:
We can keep a future<> for the background work.
The stop() calls should return a ready future anyhow (in non-debug mode), no?
since the main_compaction_group is known to be empty.